src/main/java/com/thoughtworks/aceleradora/controladores/PedidoControlador.java

```diff
@@ -59,8 +58,41 @@ public String visualizarPedido(@PathVariable("id") Long id, Model modelo, Breadc
                 .pagina("Visualizar Pedido", "/pedidos");
 
         String nomePedido = pedidoServico.encontraUm(id).getNome();
+        List<PedidoProdutoProdutor> pedidoProdutoProdutores = pedidoServico.encontraUm(id).getPedidosProdutosProdutores();
+
+        HashMap listaTotalPorProduto = new HashMap();
+
+        HashMap listaTotalPorProdutor = new HashMap();
```

Comment:
Assim como as listas, os mapas são estruturas genéricas, onde devemos especificar a qual tipo a estrutura se aplica. Falando de código isso significa usar o operador de diamantes (`<>`) para definir qual tipo de dados será armazenado no mapa. Diferentemente da lista, os mapas recebem dois parâmetros de tipo: um para chave, outro para o valor.

Como aqui estamos agrupando um total pelo id dos produtores, seria um mapa de `Long` para `BigDecimal`. Logo:

```java
Map<Long, BigDecimal> listaTotalPorProdutor = new HashMap<>();
```

Se a gente não declara os tipos do nosso mapa, ele tratará tudo como `Object`, o que é péssimo, já que somos forçados a fazer casting e perdemos a precisão dos tipos e a segurança do compilador.